File: 백준/Silver/16967. 배열 복원하기/배열 복원하기.py
import sys

input = sys.stdin.readline

def main():
    arr1 = [[arr[r][c] for c in range(w)] for r in range(h)]
    arr2 = [[arr[r][c] for c in range(y, y + w)] for r in range(x, x + h)]
    
    for r in range(x, h):
        for c in range(y, w):
            arr1[r][c] = arr2[r - x][c - y] - arr1[r - x][c - y]
    
    for rr in arr1:
        print(*rr)

h,w,x,y = map(int, input().split())
arr = [[int(i) for i in input().split()] for _ in range(h + x)]
main()

File: 백준/Silver/16967. 배열 복원하기/배열 복원하기.js
const fs = require('fs');

const [hw, ..._arr] = fs.readFileSync('dev/stdin').toString().trim().split("\n");
const [h,w,x,y] =hw.split(" ").map(Number);
const arr = _arr.map(r => r.split(" ").map(Number));

const main = () => {
    const arr1 = arr.slice(0, h).map(r => r.slice(0, w));
    const arr2 = arr.slice(x, x + h).map(r => r.slice(y, y + w));
    
    for(let r = x; r < h; r++){
        for(let c = y; c < w; c++){
            arr1[r][c] = arr2[r - x][c - y] - arr1[r - x][c - y]
        }
    }
    
    for(let rr of arr1){
        console.log(...rr);
    }
}
main()

